openhands/server/routes/settings.py

```diff
@@ -18,24 +18,27 @@
 )
 
 
-@app.get('/settings')
-async def load_settings(request: Request) -> Settings | None:
+@app.get('/settings', response_model=Settings)
+async def load_settings(request: Request) -> Settings:
     try:
         settings_store = await SettingsStoreImpl.get_instance(
             config, get_user_id(request)
         )
         settings = await settings_store.load()
         if not settings:
-            return JSONResponse(
-                status_code=status.HTTP_404_NOT_FOUND,
-                content={'error': 'Settings not found'},
-            )
+            raise ValueError('Settings not found')
         return settings
+    except ValueError as e:
+        logger.warning(str(e))
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail='Settings not found',
```

Comment:
This assumes any `ValueError` must be the error you threw at L29, which might be true now, but can evolve quickly.

How about throwing an HTTPException at L29, and here you could do

```python
    except HTTPException:
        raise
```